File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.ts
function xorAfterQueries(nums: number[], queries: number[][]): number {
    const MOD: bigint = 1000000007n;
    for (const [l, r, k, v] of queries) {
        for (let i: number = l; i <= r; i += k) {
            nums[i] = Number((BigInt(nums[i]) * BigInt(v)) % MOD);
        }
    }
    return nums.reduce((acc: number, num: number) => acc ^ num, 0);
}

File: 3653-xor-after-range-multiplication-queries-i/3653-xor-after-range-multiplication-queries-i.java
class Solution {
    private static final int MOD = 1000000007;
    public int xorAfterQueries(int[] nums, int[][] queries) {
        for (int[] query : queries) {
            int left = query[0], right = query[1], step = query[2], value = query[3];
            for (int index = left; index <= right; index += step) {
                nums[index] = (int) ((long) nums[index] * value % MOD);
            }
        }
        int answer = 0;
        for (int num : nums) {
            answer ^= num;
        }
        return answer;
    }
}